### ISSUE TITLE
About pytorch-lightning version

### ISSUE THREAD

**Issue by PhilWallace** [Learner0x5a/kTrans-release#1]
Thanks for the good work!

I tried the code but found that "from lightning import LightningModule" does not work. I changed it to "from pytorch_lightning import LightningModule" and it worked for me. Not sure if it is a bug or not.

**Comment by Learner0x5a** [comment]
Thanks for your comment!

PyTorch Lightning has been renamed to Lightning after 2.0, while PyTorch Lightning can still be installed as a standalone framework for backwards compatibility with Lightning 2.0.

For more details, please refer to [https://github.com/Lightning-AI/pytorch-lightning/discussions/16688](url)

**Comment by PhilWallace** [comment]
> Thanks for your comment!
> 
> PyTorch Lightning has been renamed to Lightning after 2.0, while PyTorch Lightning can still be installed as a standalone framework for backwards compatibility with Lightning 2.0.
> 
> For more details, please refer to [https://github.com/Lightning-AI/pytorch-lightning/discussions/16688](url)

Thanks for the reply!

May I ask what is the similarity computation method used for kTrans?

I implemented kTrans to generate function embeddings, and use cosine similarity to calculate distance:

`import torch.nn.functional as F`
`...`
`F.cosine_similarity(vector1, vector2, dim=1)`

vector1 and vector2 are the generated embeddings.

However, I found that the similarity tends to be very high. Particularly, two completely unrelated functions can have similarity above 0.9. Is it the problem with my similarity computation method? What is the similarity computation method used in the paper, or what is the recommended similarity metric?

Many thanks!

**Comment by Learner0x5a** [comment]
> > Thanks for your comment!
> > PyTorch Lightning has been renamed to Lightning after 2.0, while PyTorch Lightning can still be installed as a standalone framework for backwards compatibility with Lightning 2.0.
> > For more details, please refer to [https://github.com/Lightning-AI/pytorch-lightning/discussions/16688](url)
> 
> Thanks for the reply!
> 
> May I ask what is the similarity computation method used for kTrans?
> 
> I implemented kTrans to generate function embeddings, and use cosine similarity to calculate distance:
> 
> `import torch.nn.functional as F` `...` `F.cosine_similarity(vector1, vector2, dim=1)`
> 
> vector1 and vector2 are the generated embeddings.
> 
> However, I found that the similarity tends to be very high. Particularly, two completely unrelated functions can have similarity above 0.9. Is it the problem with my similarity computation method? What is the similarity computation method used in the paper, or what is the recommended similarity metric?
> 
> Many thanks!

We use dot product in kTrans and jTrans, which should be equivalent to cosine similarity. You can check the code [here](https://github.com/vul337/jTrans/blob/main/fasteval.py).
While this problem could have multiple reasons such as  the calculation method of similarity, architecture or code optimization, etc.

**Comment by PhilWallace** [comment]
> > > Thanks for your comment!
> > > PyTorch Lightning has been renamed to Lightning after 2.0, while PyTorch Lightning can still be installed as a standalone framework for backwards compatibility with Lightning 2.0.
> > > For more details, please refer to [https://github.com/Lightning-AI/pytorch-lightning/discussions/16688](url)
> > 
> > 
> > Thanks for the reply!
> > May I ask what is the similarity computation method used for kTrans?
> > I implemented kTrans to generate function embeddings, and use cosine similarity to calculate distance:
> > `import torch.nn.functional as F` `...` `F.cosine_similarity(vector1, vector2, dim=1)`
> > vector1 and vector2 are the generated embeddings.
> > However, I found that the similarity tends to be very high. Particularly, two completely unrelated functions can have similarity above 0.9. Is it the problem with my similarity computation method? What is the similarity computation method used in the paper, or what is the recommended similarity metric?
> > Many thanks!
> 
> We use dot product in kTrans and jTrans, which should be equivalent to cosine similarity. You can check the code [here](https://github.com/vul337/jTrans/blob/main/fasteval.py). While this problem could have multiple reasons such as the calculation method of similarity, architecture or code optimization, etc.

Noted with thanks! I will try your code as well. : )
